src/Microsoft.VisualBasic/tests/VB/SpecialDirectoriesTests.vb

```diff
@@ -0,0 +1,77 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+Option Explicit On
+Option Strict On
+
+Imports System.Environment
+Imports Microsoft.VisualBasic.FileIO
+Imports Xunit
+Namespace Microsoft.VisualBasic.Tests.VB
+    Public NotInheritable Class SpecialDirectoriesTests
+        Private Shared ReadOnly Separators() As Char = {
+            IO.Path.DirectorySeparatorChar,
+            IO.Path.AltDirectorySeparatorChar
+            }
+
+        Private Shared Sub CheckSpecialFolder(folder As SpecialFolder, getSpecialDirectory As Func(Of String))
+            If Environment.GetFolderPath(folder) = "" Then
+                Assert.Throws(Of IO.DirectoryNotFoundException)(Function() getSpecialDirectory.Invoke)
```

Comment:
Please remove `Function()`:
```
Assert.Throws(Of DirectoryNotFoundException)(getSpecialDirectory)
```

Reply:
If I remove Function I get an error below and if I remove .Invoke it does not throw exception.
![image](https://user-images.githubusercontent.com/12789369/48499917-766eb680-e7ee-11e8-93c1-dc81c1b64396.png)
![image](https://user-images.githubusercontent.com/12789369/48499952-8d150d80-e7ee-11e8-8568-0ca219f0bc14.png)

Reply:
I will need to commit to test on Unix to make sure the code below throws exception without .Invoke
```
        Private Shared Sub CheckSpecialFolder(folder As SpecialFolder, getSpecialDirectory As Func(Of Object))
            Dim LocalFolderPath As String = Environment.GetFolderPath(folder)
            If LocalFolderPath = "" Then
                Assert.Throws(Of IO.DirectoryNotFoundException)(getSpecialDirectory)
            Else
                Assert.Equal(LocalFolderPath.TrimEnd(Separators), getSpecialDirectory().ToString.TrimEnd(Separators))
            End If
        End Sub
```

Reply:
I'd overlooked the need to convert the result of `getSpecialDirectory` back to a `String`. Given that, your original approach of using `Function() getSpecialDirectory()` with `getSpecialDirectory As Func(Of String)` seems better than converting the result to a `String` later. Sorry for missing that. Please revert.